contracts/sophia.md

```diff
@@ -876,13 +876,13 @@ Data is encoded in memory as follows:
   - Tuples are encoded with one word per component (stored left-to-right) where
     each word is either an unboxed value or a pointer.
   - Lists are encoded by an unboxed -1 word for the empty list and an encoded
-    pair of the head and the tail for a cons cell.
+    pair (i.e. tuple of two components) of the head and the tail for a cons cell.
   - Records are encoded as a tuple of field values.
     - The order of values is specified as the order the fields appear in the record definition.
   - Maps are encoded as a list of key-value pairs. ***This is subject to change***
-  - The order of values is unspecified. For instance, in the encoding of a pair
-    of boxed values, the three cells (first component, second component, and
-    pair cell) can appear in any order in the encoded binary.
+    - The order of values is unspecified. For instance, in the encoding of a pair
```

Comment:
I am not confident about the content though it seems to be subpoint of maps point.

Reply:
Perhaps it should read:
"The position of a value in the binary is unspecified."
I think what we are trying to convey is that:
The tuple (17, 42) might be laid out in memory so that the addresses of components is unordered e.g. the tuple might be at address 10 and the content at address 8 and 13:
address: value
  8: 17
  9: xx
10: 8
11: 13
12: xx
13: 17

Reply:
@happi Thanks, your comment helped me in reading it better.

> The tuple (17, 42) ...

I understand that the binary that encodes a tuple with two components has to have the two components - either unboxed values or pointers - "stored left-to-right" *adjacently*. If the two components are boxed, the two pointers are adjacent while the order of the three pointers (i.e. location of the tuple, location of the first component, location of the second component) can be whatever.

17 and 42 are unboxed so I expect they must be adjacent (and in order) in memory (because part of a tuple with two components).

Reply:
I will update this branch in the following days.

Reply:
This part has been completely rewritten in https://github.com/aeternity/protocol/pull/207.